linker.go

```diff
@@ -185,12 +185,54 @@ func fixupAndValidate(insns asm.Instructions) error {
 			return fmt.Errorf("instruction %d: map %s: %w", iter.Index, ins.Reference(), asm.ErrUnsatisfiedMapReference)
 		}
 
+		if err := fixupKfunc(ins); err != nil {
+			return fmt.Errorf("fixing up kfunc: %w", err)
+		}
+
 		fixupProbeReadKernel(ins)
 	}
 
 	return nil
 }
 
+// fixupKfunc sets Instruction.Constant to the running kernels btf id
+// of the btf.Func in the instructions Metadata.
+func fixupKfunc(ins *asm.Instruction) error {
+	if !ins.IsKfuncCall() {
+		return nil
+	}
+
+	// check meta, if no meta return err
+	kfm, _ := ins.Metadata.Get(kfuncMeta{}).(*btf.Func)
+	if kfm == nil {
+		return fmt.Errorf("kfunc call has no kfuncMeta")
+	}
+
+	s, err := btf.LoadKernelSpec()
```

Comment:
This is so expensive that we either need to land https://github.com/cilium/ebpf/pull/948 or refactor this into a separate loop over `insns asm.Instructions`, similar to https://github.com/cilium/ebpf/blob/5ea5368f9823c13e71146a6c2cca84ad60a5e26a/btf/ext_info.go#L137-L194

Reply:
fixupKfunc is called within a loop over `insns` anyway, I could pull out the kernel spec load out of the function and pass the spec as an arg. That way we are not bound to another PR landing first.

Reply:
The problem is that we don't want to load the kernel spec unconditionally, since it takes 70ms! That's why I suggested a separate function with the weird double loop setup. Agreed that we should not block this on Spec.Copy improvements.